src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java

```diff
@@ -26,35 +41,69 @@
 import com.hazelcast.util.UuidUtil;
 import com.netflix.appinfo.ApplicationInfoManager;
 import com.netflix.appinfo.CloudInstanceConfig;
-import com.netflix.appinfo.MyDataCenterInstanceConfig;
+import com.netflix.appinfo.DataCenterInfo;
 import com.netflix.appinfo.EurekaInstanceConfig;
 import com.netflix.appinfo.InstanceInfo;
-import com.netflix.appinfo.DataCenterInfo;
+import com.netflix.appinfo.MyDataCenterInstanceConfig;
 import com.netflix.appinfo.providers.EurekaConfigBasedInstanceInfoProvider;
 import com.netflix.config.DynamicPropertyFactory;
 import com.netflix.discovery.DefaultEurekaClientConfig;
 import com.netflix.discovery.DiscoveryClient;
 import com.netflix.discovery.EurekaClient;
 import com.netflix.discovery.shared.Application;
 
-import java.io.IOException;
-import java.net.InetAddress;
-import java.net.URL;
-import java.net.UnknownHostException;
-import java.util.List;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Map;
-import java.util.Properties;
-import java.util.concurrent.TimeUnit;
-
-import static com.hazelcast.eureka.one.EurekaOneProperties.EUREKA_ONE_SYSTEM_PREFIX;
-import static com.hazelcast.eureka.one.EurekaOneProperties.NAMESPACE;
-import static com.hazelcast.eureka.one.EurekaOneProperties.SELF_REGISTRATION;
-
 class EurekaOneDiscoveryStrategy
         extends AbstractDiscoveryStrategy {
 
+    static final class EurekaOneDiscoveryStrategyBuilder {
+        private EurekaClient eurekaClient;
+        private ApplicationInfoManager applicationInfoManager;
+        private DiscoveryNode discoveryNode;
+        private ILogger logger = new NoLogFactory().getLogger(EurekaOneDiscoveryStrategy.class.getName());
+        private Map<String, Comparable> properties = Collections.<String, Comparable>emptyMap();
+        private boolean clientMode;
+
+        public EurekaOneDiscoveryStrategyBuilder setEurekaClient(final EurekaClient eurekaClient) {
+            this.eurekaClient = eurekaClient;
+            if (eurekaClient != null) {
+                this.applicationInfoManager = eurekaClient.getApplicationInfoManager();
```

Comment:
I have one remaining concern, regarding this assignment, since Hazelcast Eureka client also tries to register client with UP status, or DOWN when closing, I believe our client should not change app status when client is injected. It would be proper that injector context to control application UP/DOWN or other states.
@tjuchniewicz What do you think?

Reply:
You are right @lazerion ! I will fix this.

Reply:
@lazerion Please take a look at my proposed solution.

Reply:
@tjuchniewicz I am working on your branch for the updates.

Reply:
OK. Thank you.